### ISSUE TITLE
Word Imageability

### ISSUE THREAD

**Issue by StephanAkkerman** [StephanAkkerman/FluentAI#9]
We want to give priority to words that are easy to imagine so that thinking about / visualizing the mnemonic becomes easier.

Related datasets:
- https://github.com/mkasu/imageabilitycorpus ([paper](https://www.marc-kastner.com/media/pages/publications/2020-mtap/feb8c98c61-1661315763/preprint.pdf))
- Kuperman age of acquisition (part of transphoner data)
- MRC dataset (transphoner: mrc2.csv)

We got 3 datasets that cover the imageability rating:
- mrc2.csv -> 4797 words
- corpus_icmr2020.csv -> 119 words
- corpus_mipr2021.json -> 2430 words

Related papers:
- https://aclanthology.org/P14-1024.pdf / https://github.com/ytsvetko/metaphor
- https://arxiv.org/pdf/1807.02903

Option 1:
Use the approach of the paper (see image below)

Option 2:
Create a model ourselves like: 
- https://aclanthology.org/W18-3028.pdf
- https://github.com/mkasu/imageabilityestimation
- https://github.com/swsiwu/composition_and_deformance ([paper](https://aclanthology.org/2023.wnu-1.16.pdf))

Could use datasets like:
- https://github.com/babylonhealth/fastText_multilingual
- https://www.researchgate.net/publication/258061778_Concreteness_ratings_for_40_thousand_generally_known_English_word_lemmas

![image](https://github.com/StephanAkkerman/FluentAI/assets/45365128/ca5137b2-4d20-4736-85c6-7391eb491e50)

**Comment by StephanAkkerman** [comment]
Kuperman corpus: https://link.springer.com/article/10.3758/s13428-012-0210-4

**Comment by StephanAkkerman** [comment]
https://websites.psychology.uwa.edu.au/school/mrcdatabase/uwa_mrc.htm

**Comment by StephanAkkerman** [comment]
See: https://github.com/samzhang111/mrc-psycholinguistics/blob/master/wordmodel.py for column names

**Comment by StephanAkkerman** [comment]
UWN = universal wordnet

**Comment by StephanAkkerman** [comment]
Other interesting and related paper: [Estimating the imageability of words by mining visual
characteristics from crawled image data](https://www.marc-kastner.com/media/pages/publications/2020-mtap/feb8c98c61-1661315763/preprint.pdf)

**Comment by StephanAkkerman** [comment]
Data of English multiword expressions: https://github.com/ejmuraki/multiword-exp/tree/main

**Comment by StephanAkkerman** [comment]
More imageability: https://figshare.com/articles/dataset/Imageability_ratings_Cortese_Fugett_2004_/7492475

**Comment by StephanAkkerman** [comment]
MegaHR dataset is built using ML, so we will not be using it

**Comment by StephanAkkerman** [comment]
Concreteness Brysbeart et al: https://github.com/ArtsEngine/concreteness

**Comment by StephanAkkerman** [comment]
cc.en.300.bin embeddings:

Model Performances:
                       Model       MSE  R2 Score
0    Linear Regression (OLS)  0.037966 -0.069961
1           Ridge Regression  0.037340 -0.052302
2  Support Vector Regression  0.038995 -0.098944
3              Random Forest  0.036048 -0.015893
4          Gradient Boosting  0.036627 -0.032206
5                    XGBoost  0.042061 -0.185353
6                   LightGBM  0.038032 -0.071799

Best Model: RandomForestRegressor with MSE of 0.0360

**Comment by StephanAkkerman** [comment]
Model Performances:
                       Model       MSE  R2 Score
0    Linear Regression (OLS)  0.037692 -0.062227
1           Ridge Regression  0.036424 -0.026496
2  Support Vector Regression  0.037673 -0.061677
3              Random Forest  0.035481  0.000086
4          Gradient Boosting  0.036025 -0.015241
5                    XGBoost  0.041593 -0.172165
6                   LightGBM  0.037145 -0.046799

Best Model: RandomForestRegressor with MSE of 0.0355

**Comment by StephanAkkerman** [comment]
Gensim api.load("fasttext-wiki-news-subwords-300")

Model Performances:
                       Model       MSE  R2 Score
0    Linear Regression (OLS)  0.017588  0.504339
1           Ridge Regression  0.017382  0.510155
2  Support Vector Regression  0.017559  0.505167
3              Random Forest  0.018510  0.478355
4          Gradient Boosting  0.016864  0.524734
5                    XGBoost  0.017417  0.509160
6                   LightGBM  0.015981  0.549622

Best Model: LGBMRegressor with MSE of 0.0160

**Comment by StephanAkkerman** [comment]
When using the saved keyvectors from "models/fasttext.model"

Model Performances:
                       Model       MSE  R2 Score
0    Linear Regression (OLS)  0.037492 -0.056581
1           Ridge Regression  0.037001 -0.042754
2  Support Vector Regression  0.037938 -0.069170
3              Random Forest  0.036353 -0.024493
4          Gradient Boosting  0.035996 -0.014428
5                    XGBoost  0.042279 -0.191502
6                   LightGBM  0.037339 -0.052287

Best Model: GradientBoostingRegressor with MSE of 0.0360

**Comment by StephanAkkerman** [comment]
Either the embeddings or eval code is wrong, because the keyvectors should be the same as the model loaded from the api

**Comment by StephanAkkerman** [comment]
Fixed
